javascript-modules-library/src/core/server/components/render/RenderChildren.tsx

```diff
@@ -0,0 +1,58 @@
+import type { JSX } from "react";
+import { Render } from "./Render.js";
+import { useServerContext } from "../../hooks/useServerContext.js";
+import { AddContentButtons } from "../AddContentButtons.js";
+import { getChildNodes } from "../../utils/jcr/getChildNodes.js";
+import type { JCRNodeWrapper } from "org.jahia.services.content";
+
+/** Renders the children of the current node, and "Add content" buttons afterwards. */
+export function RenderChildren({
+  view,
+  pagination,
+  filter = "jnt:content",
+}: {
+  /** View to use when rendering the children. */
+  view?: string | undefined;
+  /**
+   * Pagination parameters:
+   *
+   * - `{ count: number; start?: number }` to specify the number of children to display and the
+   *   starting index (defaults to 0).
+   * - `{ count: number; page: number }` to specify the number of children to display and the page
+   *   number.
+   *
+   * If not provided, all children will be displayed.
+   */
+  pagination?: { count: number; start?: number } | { count: number; page: number };
+  /**
+   * Filter to apply to the children:
+   *
+   * - A string to filter by node type.
+   * - A function to filter by custom logic.
+   *
+   * @default "jnt:content"
+   */
+  filter?: string | ((node: JCRNodeWrapper) => boolean);
```

Comment:
Could that be useful/relevant to have both the parent node and the child in the parameters of the filtering function?

Reply:
You can use `node.getParent()` to get the parent, it's a complete JCR node